### PR TITLE
Support batching sync push requests

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/4085/support-batching-of-sync-push-requests

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarMeasurement.kt [hunk 1]
```diff
@@ -91,6 +91,17 @@ data class BloodSugarMeasurement(
     @Query("SELECT * FROM BloodSugarMeasurements WHERE syncStatus = :status")
     fun withSyncStatus(status: SyncStatus): List<BloodSugarMeasurement>
 
+    @Query("""
+      SELECT * FROM bloodsugarmeasurements
+      WHERE syncStatus = :syncStatus
+      LIMIT :limit OFFSET :offset
+    """)
+    fun withSyncStatusBatched(
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarMeasurement.kt:99]: The naming convention for other syncStatus batched is `dao#recordsWithSyncStatusBatched`. Can you add the same here as well. It'll be consistent throughout

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good overall. Left one comment PTAL

File: app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarMeasurement.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarMeasurement.kt:99]: Good point. Will do this.

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/bloodsugar/BloodSugarMeasurement.kt]: Done.

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**